File: Java/Commandes/src/Commande.java
import java.time.LocalDate;
import java.util.ArrayList;

public class Commande {
    private static int id = 0;
    private LocalDate date;
    private boolean type = false; // 1 = express | 0 = normale
    private boolean payee;
    private Client client;

    Commande(LocalDate date, boolean type, boolean payee, Client client) {
        id++;
        this.date = date;
        this.type = type;
        this.payee = payee;
        this.client = client;
    }

    int GetID() {
        return id;
    }

    LocalDate GetDate() {
        return date;
    }

    void SetDate(LocalDate date) {
        this.date = date;
    }

    boolean GetType() {
        return type;
    }

    void SetType(boolean type) {
        type = type;
    }

    boolean GetPyee() {
        return payee;
    }

    void SetPayee(boolean payee) {
        this.payee = payee;
    }

    Client GetClient() {
        return client;
    }

    void SetPayee(Client client) {
        this.client = client;
    }

    double calculTotalComande(ArrayList<LigneCommande> ligneCommandeList) {
        double totale = 0;
        for (LigneCommande ligneCommande : ligneCommandeList) {
            totale = totale + ligneCommande.calculTotalProduit();
        }
        return totale;
    }
}

File: S1/php/Sandbox/Gestion_02/classes/commande.php
<!-- This class wont be touched from the Interface :) -->
<?php
class Commande
{
    private $id;
    private $dateCom;
    private $type;
    private $payee;
    private $client;

    private static $commandeNumber = 0;

    function __construct($type, $payee, $client)
    {
        $this->id = self::$commandeNumber;
        $this->dateCom = date("Y-m-d H:i:s");
        $this->type = $type;
        $this->payee = $payee;
        $this->client = $client;
        self::$commandeNumber++;
    }

    function __destruct()
    {
        self::$commandeNumber--;
    }

    public function getID()
    {
        return $this->id;
    }

    public function getDatCom()
    {
        return $this->dateCom;
    }
    public function setDatCom($dateCom)
    {
        $this->dateCom = $dateCom;
    }

    public function getType()
    {
        return $this->type;
    }
    public function setType($type)
    {
        $this->type = $type;
    }

    public function getPayee()
    {
        return $this->payee;
    }
    public function setPayee($payee)
    {
        $this->payee = $payee;
    }

    public function getClient()
    {
        return $this->client;
    }
    public function setClient($client)
    {
        $this->client = $client;
    }
}
